src/app/store/reducers/compose-mail.reducers.ts

```diff
@@ -204,6 +239,8 @@ export function reducer(state: ComposeMailState = { drafts: {} }, action: Compos
           }
           return true;
         });
+      const isProcessingAttachments = !!state.drafts[action.payload.draftId].attachments.find(attachment => attachment.inProgress);
```

Comment:
this chunk of code is repeated on many places, maybe we should put it in a private method.